src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -470,6 +473,9 @@ public Course retrieveCourseOverExerciseGroupOrCourseId(Exercise exercise) {
     public void enrollUserForCourseOrThrow(User user, Course course) {
         authCheckService.checkUserAllowedToEnrollInCourseElseThrow(user, course);
         userService.addUserToGroup(user, course.getStudentGroupName(), Role.STUDENT);
+        if (course.getLearningPathsEnabled()) {
+            learningPathService.generateLearningPathForUser(course, user);
```

Comment:
There are multiple ways for students to join a course. If they do not enroll by themselves, the learning path won't we generated. Maybe you have to go for a lazy approach here and create the learning when the user accesses the course for the first time, for example.

Reply:
This is now part of
- #6978: Generate LP when student requests LP for the first time
- #6976: Allows instructors to generate LPs when missing